### PR TITLE
Use optional cache control headers for requests

### PR DESCRIPTION
**Pull request by royrutto**

### CODE REVIEW

File: src/cljs/milia/api/io.cljs [hunk 1]
```diff
@@ -54,13 +54,16 @@
...
-  ([token get-crsftoken?]
-     (let [headers (when (and token (not (blank? token)))
-                     {"Authorization" (str "TempToken " token)})]
-       (if-let [crsf-token (and get-crsftoken? (cks/get "csrftoken"))]
-         (assoc headers "X-CSRFToken" crsf-token)
-         headers))))
+  "Builds request headers for the HTTP request by adding
+  Authorization, X-CSRFToken and Cache-control headers where necessary"
+  [& {:keys [token get-crsftoken? must-revalidate?]}]
+    (let [Authorization #(when (and token (not (blank? token)))
```
Comment (pld) [inline src/cljs/milia/api/io.cljs]: don't need the first conjunct, `(blank? nil) ;=> true`

File: src/cljs/milia/api/io.cljs [hunk 2]
```diff
@@ -54,13 +54,16 @@
...
+                           (assoc % "Authorization" (str "TempToken " token)))
+          Cache-control #(when must-revalidate?
+                          (assoc % "Cache-control" "must-revalidate"))
+          X-CSRFToken #(when-let [crsf-token (and get-crsftoken? (cks/get "csrftoken"))]
+                        (assoc % "X-CSRFToken" crsf-token))]
+      (apply merge ((juxt Authorization Cache-control X-CSRFToken)) {})))
```
Comment (pld) [inline src/cljs/milia/api/io.cljs]: what does the `merge` do here?

Comment (royrutto) [inline src/cljs/milia/api/io.cljs]: [juxt](https://clojuredocs.org/clojure.core/juxt) returns a list of maps which need to be merged into one map e.g.

```clojure
[{"Authorization" "TempToken a1b2c3"}
 {"Cache-control" "must-revalidate"}
 {"X-CSRFToken" "x8y9z0"}]
```
needs to look like:

```clojure
{"Authorization" "TempToken a1b2c3"
 "Cache-control" "must-revalidate"
 "X-CSRFToken" "x8y9z0"}
```

Comment (pld) [inline src/cljs/milia/api/io.cljs]: Cool, thanks

**Pull request merged**